terraform/aws/scenarios/ssrf-iam-breach/main.tf

```diff
@@ -418,3 +420,57 @@ output "public_dns" {
   description = "List of public DNS names assigned to the instances. For EC2-VPC, this is only available if you've enabled DNS hostnames for your VPC"
   value       = aws_instance.web-app-instance.public_dns
 }
+
+resource "aws_flow_log" "demo-vpc" {
+  vpc_id          = "${aws_vpc.demo-vpc.id}"
+  iam_role_arn    = "<iam_role_arn>"
+  log_destination = "${aws_s3_bucket.demo-vpc.arn}"
+  traffic_type    = "ALL"
+
+  tags = {
+    GeneratedBy      = "Accurics"
+    ParentResourceId = "aws_vpc.demo-vpc"
+  }
+}
+resource "aws_s3_bucket" "demo-vpc" {
```

Comment:
**Tenable.cs remediation**
 
**[HIGH]: Ensure at-rest server side encryption (SSE) is enabled for AWS S3 Buckets**
 
**Vulnerability**
AWS S3 Buckets do not have at-rest server side encryption enabled to protect sensitive data.
 
**Remediation**
Server-side encryption protects data at rest. Amazon S3 encrypts each object with a unique key. As an additional safeguard, it encrypts the key itself with a key that it rotates regularly. Amazon S3 server-side encryption uses one of the strongest block ciphers available to encrypt your data, 256-bit Advanced Encryption Standard (AES-256).
 In Terraform - 
 Add the 'server_side_encryption_configuration' block to ensure server side encryption is enabled.

source: tenablecs-